File: venda/projeto_vendas_3/cliente-javascript/cliente.js
const axios = require("axios");
const prompt = require("prompt-sync")();

const BASE_URL = "http://localhost:8080/api";

async function listarItens() {
    const resposta = await axios.get(`${BASE_URL}/itens`);
    console.log(resposta.data);
}

async function buscarItem() {
    const id = prompt("ID do item: ");

    const resposta = await axios.get(
        `${BASE_URL}/itens/${id}`
    );

    console.log(resposta.data);
}

async function calcularTotal() {
    const id = prompt("ID do item: ");
    const quantidade = prompt("Quantidade: ");

    const resposta = await axios.get(
        `${BASE_URL}/total`,
        {
            params: {
                id: id,
                quantidade: quantidade
            }
        }
    );

    console.log("Total:", resposta.data);
}

async function listarClientes() {
    const resposta = await axios.get(
        `${BASE_URL}/clientes`
    );

    console.log(resposta.data);
}

async function listarPedidos() {
    const resposta = await axios.get(
        `${BASE_URL}/pedidos`
    );

    console.log(resposta.data);
}

async function comprarItem() {

    const clienteId = Number(prompt("ID do cliente: "));
    const itemId = Number(prompt("ID do item: "));
    const quantidade = Number(prompt("Quantidade: "));

    const resposta = await axios.post(
        `${BASE_URL}/comprar`,
        {
            clienteId,
            itemId,
            quantidade
        }
    );

    console.log(resposta.data);
}

async function menu() {

    while (true) {

        console.log("\n===== CLIENTE JAVASCRIPT =====");
        console.log("1 - Listar itens");
        console.log("2 - Buscar item");
        console.log("3 - Calcular total");
        console.log("4 - Listar clientes");
        console.log("5 - Listar pedidos");
        console.log("6 - Comprar item");
        console.log("0 - Sair");

        const opcao = prompt("Escolha: ");

        try {

            if (opcao === "1")
                await listarItens();

            else if (opcao === "2")
                await buscarItem();

            else if (opcao === "3")
                await calcularTotal();

            else if (opcao === "4")
                await listarClientes();

            else if (opcao === "5")
                await listarPedidos();

            else if (opcao === "6")
                await comprarItem();

            else if (opcao === "0")
                break;

            else
                console.log("Opção inválida.");

        } catch (erro) {
            console.log("Erro:", erro.message);
        }
    }
}

menu();

File: venda/projeto_vendas_3/cliente-python/cliente.py
import requests

BASE_URL = "http://localhost:8080/api"


def listar_itens():
    resposta = requests.get(f"{BASE_URL}/itens")
    print("\nItens disponíveis:")
    print(resposta.json())


def buscar_item():
    item_id = int(input("Digite o ID do item: "))

    resposta = requests.get(
        f"{BASE_URL}/itens/{item_id}"
    )

    print(resposta.json())


def calcular_total():
    item_id = int(input("Digite o ID do item: "))
    quantidade = int(input("Digite a quantidade: "))

    resposta = requests.get(
        f"{BASE_URL}/total",
        params={
            "id": item_id,
            "quantidade": quantidade
        }
    )

    print("\nTotal: R$", resposta.json())


def listar_clientes():
    resposta = requests.get(
        f"{BASE_URL}/clientes"
    )

    print(resposta.json())


def listar_pedidos():
    resposta = requests.get(
        f"{BASE_URL}/pedidos"
    )

    print(resposta.json())


def comprar_item():
    cliente_id = int(input("ID do cliente: "))
    item_id = int(input("ID do item: "))
    quantidade = int(input("Quantidade: "))

    dados = {
        "clienteId": cliente_id,
        "itemId": item_id,
        "quantidade": quantidade
    }

    resposta = requests.post(
        f"{BASE_URL}/comprar",
        json=dados
    )

    print("\nResultado da compra:")
    print(resposta.json())


while True:

    print("\n===== CLIENTE PYTHON =====")
    print("1 - Listar itens")
    print("2 - Buscar item")
    print("3 - Calcular total")
    print("4 - Listar clientes")
    print("5 - Listar pedidos")
    print("6 - Comprar item")
    print("0 - Sair")

    opcao = input("\nEscolha: ")

    if opcao == "1":
        listar_itens()

    elif opcao == "2":
        buscar_item()

    elif opcao == "3":
        calcular_total()

    elif opcao == "4":
        listar_clientes()

    elif opcao == "5":
        listar_pedidos()

    elif opcao == "6":
        comprar_item()

    elif opcao == "0":
        print("Encerrando...")
        break

    else:
        print("Opção inválida!")